### DISCUSSION TITLE
Is there a way to include disqus or any other custom script with  comment system on quarto blogpost?

### DISCUSSION THREAD

**Discussion by Rahuketu86** [quarto-dev/quarto-cli#2168] [Q&A]
I would like to know if it is possible to hook into the quarto comment system and provide custom.html on each post which can be enabled or disabled using comment: true / comment: false flag. I have been using jekyll/ fastpages  earlier where I could easily modify yml and post template with minimal modifications

I imagine it could be something like :
```
comments:
    disqus:
      shortname: soliloquium
```

The way to include it in Jekyll is as follows.
```
<div class="post-content e-content" itemprop="articleBody">
    {{ content | toc  }}
  </div>
  {%- if page.comments -%}
    {%- include utterances.html -%}
  {%- endif -%}
  {%- if site.disqus.shortname -%}
    {%- include disqus_comments.html -%}
  {%- endif -%}
  <a class="u-url" href="{{ page.url | relative_url }}" hidden></a>
</article>
```

Can we achieve something similar in quarto? If so how ?

**Comment by mcanouil** [comment]
It's already possible, take a look at <https://quarto.org/docs/output-formats/html-basics.html#commenting>.
The official Quarto blog is using it as an example: <https://quarto.org/docs/blog/posts/2022-07-25-feature-extensions/>.

**Reply by Rahuketu86** [reply]
I have gone through the documentation and I could understand the commenting system with utterance and giscus. However, I am still unsure of how to use the same for implementing disqus comments. Also, I am too new on quarto to understand the extension mechanism and how to write filter modifying comment system behavior. Do you know of  any minimal example somewhere which specifically deal with comment system . If I can understand it, I can try to implement it.

**Reply by mcanouil** [reply]
The part of the documentation which describe the way to include a custom HTML script is https://quarto.org/docs/output-formats/html-basics.html#includes
You would have to write the proper code block for discuss.

Regarding, the yaml enable/disable option, this will require a Lua filter.

I'll try to make a working example later.

**Reply by harrelfe** [reply]
The extensive advertisements that appear when using `disqus` makes me in favor of moving to `hypothes.is`.

**Reply by Rahuketu86** [reply]
I want to use Disqus because it provides an option for non-technical audiences like my father to use Facebook/ Twitter login to comment ( creating a GitHub account would be a stretch for them). Nonetheless, for me, I am more interested in figuring out how to make it work so would really appreciate any guidance on the same.

**Reply by mcanouil** [reply]
Pick the option you like:
- Create a format with a partial (Pandoc template framework) and a Lua filter to access yaml option to enable/disable the feature you are adding in the partial
- Use `include-after` or `include-after-body` (as answered https://github.com/quarto-dev/quarto-cli/discussions/2168#discussioncomment-3491804) to include a file with your HTML code for Discuss or inline.  
  Don't include the HTML when you don't want Discuss, or include it when you want it.
  In your case, the file to include would be `disqus_comments.html` as per your example.

**Reply by fuhrmanator** [reply]
> The extensive advertisements that appear when using `disqus` makes me in favor of moving to `hypothes.is`.

I use ad-blockers and don't see that noise from `disqus`, so I never thought about that. 

However, I'm interested in how a user who has not already installed `hypothes.is` will know it's an option to make comments at the end of a blog post. I enabled it in my Quarto blog, and it's possible (when you select text) to make annotations. However, there doesn't seem to be any hints (in the hypothesis-enabled site) that it's even possible.

**Comment by davidleitch1** [comment]
Recently I added successfully added [Isso](https://isso-comments.de) comments to my Quarto [ITK](https://www.itkservices3.com) site hosted on Netlify. Isso runs via a Cloudflare tunnel outside of Netlify. The  Javascript file,, supplied by Isso, is part of a post via including it as HTML and added via:
```
Include-after-body
```
I only mention this in case someone wants something other than Gisqus. I don't expect many comments and my readers are unlikely to have Github accounts.

It was an interesting exercise and took an amateur like me about 4 days. It's great to have but Isso is by default unmoderated and allows unregistered comments. This can be changed, email can be a required input, although not validated. Thought it was worth mentioning.
